### PR TITLE
Added subs check and parsing

### PR DESCRIPTION
**Pull request by Rocco83**
Now any string is parsed, even subs ids.
the key for the yaml has changed to keep both keys for debug.

This solve also https://github.com/RedHatSatellite/katello-attach-subscription/issues/14

@evgeni can you please check has_resource on your side?
on latest sat 6.2 seems that host_subscriptions exists, so 
if api.has_resource?(:host_subscriptions)
is true.

### CODE REVIEW

**Comment by Rocco83** [comment]
Hi,

The function has been rebased and improved.
Some tests has been done.
Code has been speeded up by moving one api call only when really needed (sys = ).
Changed --use-cache in --read-from-cache
Added sanity check
Added --verbose flag
Fixed (hopefully) indentations -- some may be still wrong, especially for DEBUG

Well... major rewrite i would say.

File: katello-attach-subscription [hunk 1]
```diff
@@ -71,13 +72,16 @@ optparse = OptionParser.new do |opts|
...
   end
-  opts.on("--cache-file=FILE", "read or write to cache file, based on --use-cache value") do |cf|
+  opts.on("--cache-file=FILE", "read or write to cache file, based on --read-from-cache value") do |cf|
     @options[:cachefile] = cf
   end
-  opts.on("-d", "--debug", "debug the script") do
+  opts.on("-v", "--verbose", "verbose output for the script") do
+    @options[:verbose] = true
+  end
+  opts.on("-d", "--debug", "debug output for the script") do
```
Comment (evgeni) [inline katello-attach-subscription:85]: I think `--debug` should just set `verbose=true` too, so you don't have to do `if verbose or debug` everywhere

File: katello-attach-subscription [hunk 2]
```diff
@@ -102,11 +106,115 @@ end
...
+def checksubs()
+  # initialize variables
+  # by default any kind of subscription is fine
+  subfiltertype = nil
+
+  # create api binding
+  api = ApipieBindings::API.new({:uri => @options[:uri], :username => @options[:user], :password => @options[:pass], :api_version => '2', :timeout => @options[:timeout]})
+
+  # If type is Hypervisor, only NORMAL subscription has to be considered
+  if @yaml[:type] = "Hypervisor"
```
Comment (evgeni) [inline katello-attach-subscription]: Where does that come from? We don't define `type` directly in the YAML, just in the individual subscription sections

File: katello-attach-subscription [hunk 3]
```diff
@@ -102,11 +106,115 @@ end
...
+    subfiltertype = "NORMAL"
+  end
+
+  # sanity check and fix of yaml file for subscription
+  # going to check if any id in the yaml do not exists, and expand the search keys
+  # sub will keep the originally requested subscriptions, sub_parsed the found one
+  puts "Subscription parsing started. Please be patient."
+  subsyamltotalentry = @yaml[:subs].count
+  @yaml[:subs].each_with_index do |sub, subcurrentcount|
+    puts " Started parsing entry #{subcurrentcount}#{subsyamltotalentry}"
```
Comment (evgeni) [inline katello-attach-subscription]: missing separator between current and total?

**Review by evgeni — CHANGES_REQUESTED** [review]

File: katello-attach-subscription (hunk 1 above)
Comment (Rocco83) [reply katello-attach-subscription:85]: Ok sounds good. Will be done shortly.

File: katello-attach-subscription (hunk 3 above)
Comment (Rocco83) [reply katello-attach-subscription]: Yes thanks

**Comment by Rocco83** [comment]
Seems that i cannot mark it somehow as resolved, but it is resolved since https://github.com/RedHatSatellite/katello-attach-subscription/pull/15/commits/3fd34e025fc65a4dc6540fcebff2353183dfd3a8
@evgeni may you please commit ? :)

**Comment by evgeni** [comment]
I think this was part of #26, closing

**Pull request closed without merging**